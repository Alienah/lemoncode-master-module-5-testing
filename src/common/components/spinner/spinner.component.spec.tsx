import React from 'react';
import { render, screen } from '@testing-library/react';
import { usePromiseTracker } from 'react-promise-tracker';
import { SpinnerComponent } from './spinner.component';

jest.mock('react-promise-tracker', () => {
  const original = jest.requireActual('react-promise-tracker');
  return {
      ...original,
      usePromiseTracker: jest.fn()
  };
});
const mockUsePromiseTracker = usePromiseTracker as jest.MockedFunction<typeof usePromiseTracker>

describe('common/components/SpinnerComponent', () => {
  afterEach(() => {
    jest.resetModules();
  })

  it('should render an empty component when promiseInProgress is false', () => {
    // Arrange
    mockUsePromiseTracker.mockImplementation(() => ({ promiseInProgress: false }));

    // Act
    const {baseElement} = render(<SpinnerComponent />);
    const modal = screen.queryByRole('presentation')

    // Assert
    expect(baseElement).toMatchSnapshot();
    expect(modal).not.toBeInTheDocument();
  });
  it('should render a modal with loader when promiseInProgress is true', () => {
    // Arrange
    mockUsePromiseTracker.mockImplementation(() => ({promiseInProgress: true}));

    // Act
    const {baseElement}= render(<SpinnerComponent />);
    const modal = screen.getByRole('presentation');
    const loader = screen.getByRole('img');
    const loaderContent = loader.querySelectorAll('span')[0];

    // Assert
    expect(baseElement).toMatchSnapshot();
    expect(modal).toBeInTheDocument();
    expect(loader).toBeInTheDocument();
    expect(loaderContent).toBeInTheDocument();
  });
});
