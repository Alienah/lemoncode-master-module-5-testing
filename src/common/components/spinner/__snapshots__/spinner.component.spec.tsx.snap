// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`common/components/SpinnerComponent should render a modal with loader when promiseInProgress is true 1`] = `
<body
  style="padding-right: 0px; overflow: hidden;"
>
  <div
    aria-hidden="true"
  />
  <div
    class="css-1hppjzv"
    role="presentation"
    style="position: fixed; z-index: 1300; right: 0px; bottom: 0px; top: 0px; left: 0px;"
  >
    <div
      aria-hidden="true"
      style="z-index: -1; position: fixed; right: 0px; bottom: 0px; top: 0px; left: 0px; background-color: rgba(0, 0, 0, 0.5);"
    />
    <div
      data-test="sentinelStart"
      tabindex="0"
    />
    <div
      aria-label="loader icon"
      class="css-1gb2dig"
      role="img"
      tabindex="-1"
    >
      <span
        class="css-1xdhyk6"
      >
        <span
          class="css-129peub"
        />
        <span
          class="css-prnvxp"
        />
        <span
          class="css-8el4s5"
        />
        <span
          class="css-1hf2v34"
        />
        <span
          class="css-tx0a7l"
        />
      </span>
    </div>
    <div
      data-test="sentinelEnd"
      tabindex="0"
    />
  </div>
</body>
`;

exports[`common/components/SpinnerComponent should render an empty component when promiseInProgress is false 1`] = `
<body>
  <div />
</body>
`;
